library/gui/tooltip/src/main/java/org/quiltmc/qsl/tooltip/api/package-info.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * <h2>The Quilt Tooltip API</h2>
+ *
+ * <p>
+ * <h3>What are the tooltip APIs?</h3>
+ * With Minecraft 1.17, a new system has been introduced through the {@link net.minecraft.item.BundleItem}: custom tooltip components.
+ * <p>
+ * Those tooltip components are done using mainly two classes:
+ * <ul>
+ *     <li>{@link net.minecraft.client.item.TooltipData} which will hold information about the tooltip, exist on both sides;</li>
+ *     <li>{@link net.minecraft.client.gui.tooltip.TooltipComponent} which will do the rendering of the tooltip, only exist on the client.</li>
+ * </ul>
+ * An item can return a custom tooltip data with its method {@link net.minecraft.item.Item#getTooltipData(net.minecraft.item.ItemStack)}.
+ * The issue is the conversion from the tooltip data to the tooltip component, the tooltip data is not aware of a conversion mechanism.
```

Comment:
Please check that I understood this correctly.
```suggestion
 * An item can return a custom tooltip data by overriding the method {@link net.minecraft.item.Item#getTooltipData(net.minecraft.item.ItemStack)}.
 * However, there is no method in Vanilla to convert custom tooltip data into a component.
```